### PR TITLE
wicked: move add_serial_console into pre_run hook

### PR DESCRIPTION
**Pull request by asmorodskyi**
move add_serial_console into pre_run hook

VR: 
https://openqa.suse.de/tests/3247991

### CODE REVIEW

File: lib/wickedbase.pm [hunk 1]
```diff
@@ -379,18 +379,21 @@ sub upload_wicked_logs {
...
+        script_run("journalctl -b -o short-precise|tail -n +2 > $logs_dir/journalctl.log");
+        script_run("wicked ifstatus --verbose all > $logs_dir/wicked_ifstatus.log 2>&1");
+        script_run("wicked show-config > $logs_dir/wicked_config.log 2>&1");
+        script_run("wicked show-xml > $logs_dir/wicked_xml.log 2>&1");
+        script_run("ip addr show > $logs_dir/ip_addr.log 2>&1");
+        script_run("ip route show table all > $logs_dir/ip_route.log 2>&1");
+        script_run("cat /etc/resolv.conf > $logs_dir/resolv.conf 2>&1");
+        script_run("cp /tmp/wicked_serial.log $logs_dir/") if $prefix eq 'post';
+        script_run("tar -C /tmp/ -cvzf $dir_name.tar.gz $dir_name");
+    };
```
Comment (cfconrad) [inline lib/wickedbase.pm]: script_run() should not raise an exception or?

File: lib/wickedbase.pm [hunk 2]
```diff
@@ -560,26 +563,25 @@ sub post_run {
...
-    record_info('Failed to upload post logs', $@, result => 'fail') unless defined $result;
+    $self->upload_wicked_logs('post');
 }
 
 sub pre_run_hook {
     my ($self) = @_;
+    if (get_var('VIRTIO_CONSOLE_NUM', 1) > 1) {
+        my $result = eval { add_serial_console('hvc1'); };
+        croak('Failed to add serial console : ' . $@) unless defined $result;
+    }
```
Comment (cfconrad) [inline lib/wickedbase.pm]: What exception you are trying to catch here?

File: lib/wickedbase.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/wickedbase.pm]: is it ok now ?

File: lib/wickedbase.pm (hunk 2 above)
Comment (asmorodskyi) [reply lib/wickedbase.pm]: you right - none , removed . and also we don't want to run this before **every** pre_run but just once

Comment (cfconrad) [reply lib/wickedbase.pm]: I think `$self->select_serial_terminal()` makes sense for every run.

File: lib/wickedbase.pm (hunk 1 above)
Comment (cfconrad) [reply lib/wickedbase.pm]: yes

**Pull request merged**